pkg/helper/client.go

```diff
@@ -40,15 +41,15 @@ const defaultGwProbeTimeout = 120 * time.Second
 const apiServerProbeTimeout = 120 * time.Second
 
 var (
+	setMutex             = &sync.Mutex{}
```

Comment:
Can we move out variables to the configurations/ package? Keep it a global, but private. Expose a structure with Synchronous getters and setters to set and read them. Does it sound good? Put all the configuration on a single space instead of sharing it between modules and put it behind a nice abstraction.